lib/tasks/update_search_trigrams.rake

```diff
@@ -1,31 +1,54 @@
 namespace :search do
 
+  def rkey(resource)
+    "rake-task-update-#{resource}-tsvectors"
+  end
+
+  def start_id(resource)
+    start_batch = if redis.exists(rkey(resource))
+      redis.get(rkey(resource))
+    else
+      # Returns either the arg or the default, 0
+      args[:start].to_i
+    end
+    start_batch
+  end
+
   desc 'Update comment tsvectors'
   task :update_comment_tsvectors, [:start, :batch_size] => :environment do |t, args|
     args.with_defaults(start: 0, batch_size: 1000)
-    Comment.find_in_batches(start: args[:start].to_i, batch_size: args[:batch_size].to_i).with_index do |comments, batch|
-      puts "Processing comment group ##{batch}"
+    redis = Redis.new(host: ENV['REDIS_URL'], db: 15)
+
+    Comment.find_in_batches(start: start_id('comment'), batch_size: args[:batch_size].to_i).with_index do |comments, batch|
```

Comment:
Are we always going to pass in a batch size now or just leave it as the default? I'd probably just use the default 1000 size tbh.

If you do use the arg make sure it's is supplied and not `nil.to_i` (0)

Reply:
`args.with_defaults` ensures no nulls come through, default is 1000.

Reply:
Means this is runnable with `rake:update_comment_tsvectors` or `rake:update_comment_tsvectors[500000,500]` or w/e

Reply:
🤯 marvellous